### ISSUE TITLE
not registerDriver when class.forName()

### ISSUE THREAD

**Issue by mayyamus** [neo4j/neo4j-jdbc#104]
Doc of JDBC: When a Driver class is loaded,it should create an instance of itself and register it with the DriverManager.
I think should put register code in static code block like other database jdbc driver,for example:
```
public class Driver extends NonRegisteringDriver implements java.sql.Driver {
	// ~ Static fields/initializers
	// ---------------------------------------------

	//
	// Register ourselves with the DriverManager
	//
	static {
		try {
			java.sql.DriverManager.registerDriver(new Driver());
		} catch (SQLException E) {
			throw new RuntimeException("Can't register driver!");
		}
	}

	// ~ Constructors
	// -----------------------------------------------------------

	/**
	 * Construct a new driver and register it with DriverManager
	 * 
	 * @throws SQLException
	 *             if a database error occurs.
	 */
	public Driver() throws SQLException {
		// Required for Class.forName().newInstance()
	}
}
```
![jdbc](https://user-images.githubusercontent.com/24735504/26885122-47292af2-4bd4-11e7-97e0-fbe27b31c8fd.png)

```
Class.forName();
DriverManager.getConnection(url,props)
```
in the JEE Container,will thow No suitable driver found for ...

Class.forName().newInstance() trigger registerDriver,webapp is run ok.

**Comment by inserpio** [comment]
Good point @mayyamus, thanks.
We need to think a bit about what's eventually wrong with the current implementation, as you aren't the first person opening an issue regarding the JEE container.
We currently suggest to solve the problem as in this comment: https://github.com/neo4j-contrib/neo4j-jdbc/issues/101#issuecomment-301517103.
But your suggestion looks to be very interesting and correct.

**Comment by glaggia-larus** [comment]
I think this is a good issue, we for sure need to implements the static block as requested from the specs.
I add this into the next milestone and I think I will implement this as soon as possible.

**Comment by glaggia-larus** [comment]
This is done and will be released in the next 3.2 version.
